plugins/usbdmx/AsyncPluginImpl.cpp

```diff
@@ -353,17 +346,21 @@ bool AsyncPluginImpl::StartAndRegisterDevice(Widget *widget, Device *device) {
 
 /*
  * @brief Signal widget removal.
- * @param device_id The device id to remove.
+ * @param state The DeviceState owning the widget to be removed.
  *
- * This is run within the main thread.
+ * This is run within the main thread and it must. The Device destructor
+ * below may cause libusb_close() to be called, which will deadlock if the
+ * hotplug event thread waits for it.
```

Comment:
I wonder if this reads a little better?
```suggestion
 * This is run within the main thread and it must be as the Device destructor
 * below may cause libusb_close() to be called, which will deadlock if the
 * hotplug event thread waits for it.
```

Reply:
Aww, and I was so proud of my forceful sounding formulation :), but sure I'll incorporate this into the next revision.

Reply:
There's nothing wrong with it forcefulness wise, just grammatically it doesn't read quite right. We could go with this instead if you want, which is closer to your original:
```suggestion
 * This must be run within the main thread. The Device destructor below may
 * cause libusb_close() to be called, which will deadlock if the hotplug event
 * thread waits for it.
```

You could always chuck in an exclamation mark! :wink:

Reply:
I've applied your suggestion and added a bit more detail around where/how things would deadlock.